### PR TITLE
Typed registers

### PR DESCRIPTION
**Pull request by johnor**

### CODE REVIEW

**Review by mkiael — APPROVED** [review]
Nice, this made the code more readable!

File: core/include/nes/core/ppu_registers.h [hunk 1]
```diff
@@ -4,6 +4,55 @@
...
 namespace n_e_s::core {
 
+template <class TypeT>
+class Register {
+public:
+    constexpr Register() = default;
+    constexpr explicit Register(const TypeT value) : value_{value} {};
+
+    [[nodiscard]] constexpr bool operator==(const Register &) const = default;
+    [[nodiscard]] constexpr bool operator!=(const Register &) const = default;
```
Comment (robinlinden) [inline core/include/nes/core/ppu_registers.h:14]: Defining == makes the compiler generate != for you (in C++20 and newer.)
>The inequality operator is automatically generated by the compiler if operator== is defined.
https://en.cppreference.com/w/cpp/language/operators

**Review by robinlinden — APPROVED** [review]
Neat! Really nice cleanup!

File: core/include/nes/core/ppu_registers.h (hunk 1 above)
Comment (johnor) [reply core/include/nes/core/ppu_registers.h:14]: Cool, didn't know that! I'll remove it

**Pull request merged**